packages/flutter_blue_plus_platform_interface/lib/src/bluetooth_msgs.dart

```diff
@@ -336,13 +343,15 @@ class BmBluetoothDescriptor {
   final DeviceIdentifier remoteId;
   final Guid serviceUuid;
   final Guid characteristicUuid;
+  final int characteristicIndex;
   final Guid descriptorUuid;
   final Guid? primaryServiceUuid;
 
   BmBluetoothDescriptor({
     required this.remoteId,
     required this.serviceUuid,
     required this.characteristicUuid,
+    required this.characteristicIndex,
```

Comment:
Should we default this to `0` and make it optional in order to avoid a breaking change?

```suggestion
    this.characteristicIndex = 0,
```

https://github.com/chipweinberger/flutter_blue_plus/issues/795#issuecomment-2461566048